File: src/main/java/ru/dkalchenko/teatime/model/Person.java
package ru.dkalchenko.teatime.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Objects;

@Getter
@Setter
@Document
@RequiredArgsConstructor
@NoArgsConstructor
public class Person {

    @Id
    private String id;

    @NonNull
    private String firstName;

    @NonNull
    private String lastName;

    @NonNull
    private String email;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Person person = (Person) o;
        return firstName.equals(person.firstName) && lastName.equals(person.lastName) && email.equals(person.email);
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstName, lastName, email);
    }
}

File: src/main/java/ru/dkalchenko/teatime/model/Person.kt
package ru.dkalchenko.teatime.model

import lombok.Getter
import lombok.NoArgsConstructor
import lombok.RequiredArgsConstructor
import lombok.Setter
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Getter
@Setter
@Document
@RequiredArgsConstructor
@NoArgsConstructor(force = true)
class Person {
    @Id
    private val id: String = null
    private val firstName: String = null
    private val lastName: String = null
    private val email: String = null
    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val person = o as Person
        return firstName == person.firstName && lastName == person.lastName && email == person.email
    }

    override fun hashCode(): Int {
        return Objects.hash(firstName, lastName, email)
    }
}
